nsxt/resource_nsxt_edge_cluster.go

```diff
@@ -8,10 +8,18 @@ import (
 	"log"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
 	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx"
 	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
 )
 
+var enableFailureDomainAllocation = []string{
+	"true",
```

Comment:
If we have three options here rather than two, I would suggest `enable`, `disable` and `no_preference` options here as well (and rename attr to `failure_domain_allocation`)

Reply:
Discussion in https://github.com/vmware/terraform-provider-nsxt/pull/957#issuecomment-1718370583 makes me uncertain if we need three options here. 
I didn't find any doc mentioning what allocation rule and behavior will be used if users don't provide one. If `no_preference` and `disable` are the same then we could use a boolean here.

Reply:
While @GraysonWu is correct, using a boolean might create a problem in case of future API changes where `no_preference` will end up having different semantics than `disable`

Reply:
I don't think behavior changes are a big concern, since this API will be replaced by policy API in future